### ISSUE TITLE
Fix Windows packaging regression

### ISSUE THREAD

**Issue by jcfr** [Slicer/Slicer#6607]
## Summary

Following the integration https://github.com/Slicer/Slicer/pull/6198, the windows packaging is failing with the following error (copied from `_CPack_Packages/win-amd64/NSIS/NSISOutput.log`):

```
warning 6001: Variable "AR_RegFlags" not referenced or never set, wasting memory!
Error setting manifest: Incompatible option
Error - aborting creation process
```

## Steps to reproduce

Attempt to package Slicer on `overload` windows factory

## Expected behavior

Successful packaging

## Environment
- Operating system: Windows

| NSIS version 3.05 | Windows 10 Pro / build 19044.2130 |
|--|--|
|![image](https://user-images.githubusercontent.com/219043/197081309-24000fa5-ae33-483a-b109-c217ee92b5fe.png)| ![image](https://user-images.githubusercontent.com/219043/197081457-4181833a-85df-4537-a465-b909ab245f80.png) |

**Comment by jcfr** [comment]
Updating the NSIS version from `3.05` to `3.08` allows to successful build the package.

This was testing using:

```
D:
cd" D:\D\P\S-0-build\Slicer-build\_CPack_Packages\win-amd64\NSIS"
"C:\Program Files (x86)\NSIS\makensis.exe" project.nsi

[...]

Output: "D:\D\P\S-0-build\Slicer-build\_CPack_Packages\win-amd64\NSIS\Slicer-5.1.0-2022-10-19-win-amd64.exe"
Install: 7 pages (448 bytes), 3 sections (3 required) (6216 bytes), 10015 instructions (280420 bytes), 11135 strings (678878 bytes), 64 language tables (21632 bytes).
Uninstall: 2 pages (192 bytes), 1 section (2072 bytes), 10387 instructions (290836 bytes), 12258 strings (1373802 bytes), 64 language tables (16256 bytes).
Datablock optimizer saved 861333 bytes (~0.3%).

Using lzma compression.

EXE header size:              175104 / 38400 bytes
Install code:                 153539 / 988058 bytes
Install data:              224271716 / 900191367 bytes
Uninstall code+data:          122801 / 128397 bytes
CRC (0xCFE13557):                  4 / 4 bytes

Total size:                224723164 / 901346226 bytes (24.9%)
```

**Comment by jcfr** [comment]
The regression was likely fixed in NSIS by kichik/nsis@41321856700eadd6a260918dc207e8dfea86c1e3  now setting to `m_target_type` to `TARGET_X86UNICODE` and avoiding the exception below from being raised (because `build_unicode` is now `true`):

```cpp
...
int CEXEBuild::SetManifest()
{
  try {
    init_res_editor();
    manifest::SPECIFICATION spec = { (manifest::flags) manifest_flags, manifest_dpiaware, manifest_dpiawareness.c_str(), manifest_lpaware, &manifest_sosl, manifest_maxversiontested.c_str() };
    string manifest = manifest::generate(manifest_comctl, manifest_exec_level, spec);

    if (manifest == "")
      return PS_OK;

    if (!build_unicode && manifest_lpaware >= manifest::lpaware_true)
      throw std::runtime_error("Incompatible option");

    ...

    // Saved directly as binary into the exe.
    res_editor->UpdateResource(MAKEINTRESOURCE(24), 1, NSIS_DEFAULT_LANG, (LPBYTE) const_cast<char*>(manifest.c_str()), (DWORD) manifest.length());
  }
  catch (exception& err) {
    ERROR_MSG(_T("Error setting manifest: %") NPRIs _T("\n"), CtoTStrParam(err.what()));
    return PS_ERROR;
  }

  return PS_OK;
}
```
Source: https://github.com/kichik/nsis/blob/v308/Source/build.cpp#L2399-L2410
